test/Transactions/Orleans.Transactions.Tests/Runners/GoldenPathTransactionTestRunner.cs

```diff
@@ -14,30 +15,40 @@ protected GoldenPathTransactionTestRunner(IGrainFactory grainFactory, ITestOutpu
 
         [SkippableTheory]
         [InlineData(TransactionTestConstants.SingleStateTransactionalGrain)]
+        [InlineData(TransactionTestConstants.DoubleStateTransactionalGrain)]
+        [InlineData(TransactionTestConstants.EightStateTransactionalGrain)]
         public virtual async Task SingleGrainReadTransaction(string transactionTestGrainClassName)
         {
             const int expected = 0;
 
             ITransactionTestGrain grain = RandomTestGrain(transactionTestGrainClassName);
-            int actual = await grain.Get();
-            Assert.Equal(expected, actual);
+            var actualResults = await grain.Get();
+            //each transaction state should all be 0 since no operation was applied yet
+            foreach (var actual in actualResults)
+            {
+                Assert.Equal(expected, actual);
+            }
         }
 
         [SkippableTheory]
         [InlineData(TransactionTestConstants.SingleStateTransactionalGrain)]
+        [InlineData(TransactionTestConstants.DoubleStateTransactionalGrain)]
+        [InlineData(TransactionTestConstants.EightStateTransactionalGrain)]
         public virtual async Task SingleGrainWriteTransaction(string transactionTestGrainClassName)
         {
             const int delta = 5;
             ITransactionTestGrain grain = RandomTestGrain(transactionTestGrainClassName);
-            int original = await grain.Get();
+            var original = await grain.Get();
             await grain.Add(delta);
-            int expected = original + delta;
-            int actual = await grain.Get();
+            var expected = original.Select(value => value += delta).ToArray();
```

Comment:
+= is a little odd here, since changes to `value` will have no effect.

Reply:
yeah it should be just +